raft/src/test/java/org/apache/kafka/raft/MockFuturePurgatory.java

```diff
@@ -0,0 +1,100 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.kafka.raft;
+
+import org.apache.kafka.common.errors.TimeoutException;
+import org.apache.kafka.common.utils.MockTime;
+
+import java.util.PriorityQueue;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.atomic.AtomicLong;
+
+public class MockFuturePurgatory<T> implements FuturePurgatory<T>, MockTime.Listener {
+    private final AtomicLong idGenerator = new AtomicLong(1L);
+    private final MockTime time;
+    private final PriorityQueue<DelayedFuture> delayedFutures = new PriorityQueue<>();
+
+    public MockFuturePurgatory(MockTime time) {
+        this.time = time;
+        time.addListener(this);
+    }
+
+    @Override
+    public void await(
```

Comment:
nit: looks more natural in this way:
```
    @Override
    public void await(CompletableFuture<T> future, 
                      long maxWaitTimeMs) {
```

Reply:
I've started preferring this form because the arguments always line up at the same indentation and it allows for a separate line for the `throws` clause. Otherwise, we end up with ugly stuff like this:

```java
void someReallyBigAndUglyFunction(BigAndUglyArg1 foo, 
                                  BigAndUglyArg2 bar) throws Exception;
```
As opposed to this:
```java
void someReallyBigAndUglyFunction(
    BigAndUglyArg1 foo, 
    BigAndUglyArg1 bar
) throws Exception;
```
I admit it's a matter of taste, so I won't start imposing this style on others.